Jenkinsfile

```diff
@@ -0,0 +1,40 @@
+#!groovy
+
+def nodeLabel = 'docker'
+
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+node(nodeLabel) {
+  stage 'Checkout'
+  checkout scm
+
+  stage 'Build'
+
+  /* Call the maven build. */
+  mvn "clean install -B"
+
+  /* Archive the test results */
+  step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
+
+  /* Archive the build artifacts */
+  step([$class: 'ArtifactArchiver', artifacts: 'target/*.hpi'])
+}
+
+/* Run maven from tool "mvn" */
+void mvn(def args) {
+    /* Get jdk tool. */
+    String jdktool = tool name: "jdk8", type: 'hudson.model.JDK'
+
+    /* Get the maven tool. */
+    def mvnHome = tool name: 'mvn'
+
+    /* Set JAVA_HOME, and special PATH variables. */
+    List javaEnv = ["PATH+JDK=${jdktool}/bin", "JAVA_HOME=${jdktool}"]
+
+    /* Call maven tool with java envVars. */
+    withEnv(javaEnv) {
+      sh "${mvnHome}/bin/mvn ${args}"
```

Comment:
Mark, in the meanwhile until infra-606, can we go ahead and merge this?  I'm happy to take a task to test/finish the multiplatform-ness when that is goes in.